src/cargo/sources/registry/mod.rs

```diff
@@ -411,6 +411,20 @@ impl<'a> RegistryDependency<'a> {
     }
 }
 
+pub enum LoadResponse {
+    /// The cache is valid. The cached data should be used.
+    CacheValid,
+
+    /// The cache is out of date. Returned data should be used.
+    Data {
+        raw_data: Vec<u8>,
+        index_version: Option<InternedString>,
```

Comment:
@Eh2406 Should this be an `InternedString` or just a `String`? I'm not really sure when it's appropriate to use `InternedString`.

For the HTTP registry case it's likely every crate could have a different `index_version` (etag or LM header).

Reply:
`InternedString` is good for when we have the same text in many places, or the text will get cloned a lot, or managing a lifetime is to painful. `&'a str` will not work here. `String` is a close call as the git index will have the same text in a lot of strings, but http will not. I think `InternedString` is reasonable.

Reply:
I changed this to a regular string.